src/openrct2/ride/Ride.cpp

```diff
@@ -7821,3 +7783,20 @@ size_t Ride::FormatNameTo(void* argsV) const
         return ft.NumBytes();
     }
 }
+
+uint64_t Ride::GetAvailableModes() const
+{
+    if (gCheatsShowAllOperatingModes)
+        return AllRideModesAvailable;
+
+    return GetRideTypeDescriptor().RideModes;
+}
+
+const RideTypeDescriptor& Ride::GetRideTypeDescriptor() const
+{
+    if (type >= std::size(RideTypeDescriptors))
+    {
+        return DummyRTD;
+    }
```

Comment:
Is this an expected scenario or a failure one? I'm just wondering if we should log an error inside this `if`.